common/datavalues2/src/columns/boolean/mod.rs

```diff
@@ -99,6 +124,8 @@ impl Column for BooleanColumn {
         unsafe {
             Arc::new(Self {
                 values: self.values.clone().slice_unchecked(offset, length),
```

Comment:
It's false cached, so we do not need to clone the original slices?

Reply:
if not cached data, we do not need to copy the data, but if data cached, it seems that we need to copy the underlying data and make `data_cached` true.